0019/README.md

```diff
@@ -152,6 +154,133 @@ Endpoint metadata get response message payload MUST be an Avro-encoded object wi
 }
 ```
 
-### Timeout and retry
+
+## Partial endpoint metadata update
+
+Completing a partial metadata update request will update or create only the endpoint metadata key-value pairs present in the request payload using [JSON Patch](https://tools.ietf.org/html/rfc6902). Metadata keys must be non-empty strings containing Latin letters, digits, and underscores only.
+
+### Partial metadata update request
+
+*Partial endpoint metadata update request* message is a [targeted message](/0003/README.md#targeted-messaging) that the client sends to the repository to patch endpoint metadata.
+
+The client MUST send partial endpoint metadata update request messages using the following NATS subject:
+```
+kaa.v1.service.{repository-service-instance-name}.epmmp.ep-metadata-patch-request
+```
+where:
+- `{repository-service-instance-name}` is the endpoint metadata repository service instance name. 
+
+The client MUST include NATS `replyTo` field to handle the response. It is RECOMMENDED to follow the subject format described in [3/ISM session affinity section](/0003/README.md#session-affinity):
+```
+kaa.v1.replica.{client-service-replica-id}.epmmp.ep-metadata-patch-response
+```
+
+Partial endpoint metadata update request message payload MUST be an [Avro-encoded](https://avro.apache.org/) object with the following schema ([0019-endpoint-metadata-patch-request.avsc](./0019-endpoint-metadata-patch-request.avsc)):
+```json
+{
+    "namespace":"org.kaaproject.ipc.epmmp.gen.v1",
+    "name":"EndpointMetadataPatchRequest",
+    "type":"record",
+    "doc":"Interservice endpoint parcial metadata update request",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"endpointId",
+            "type": "string",
+            "doc":"Identifier of the endpoint, on behalf of which metadata is requested"
+        },
+        {
+            "name":"operations",
+            "type":{
+                "type":"array",
+                "items":"EndpointMetadataPatchOperation"
+            },
+            "doc":"List of metadata operations"
+        }
+    ]
+}
+```
+
+In case the client requests to create or update a restricted EP metadata key, the server MUST return an error.
+
+
+### Partial endpoint metadata update response
+
+*Partial metadata update response* message MUST be sent by the repository in response to [Partial metadata update request message](#partial-metadata-update-request).
```

Comment:
to \[Partial **endpoint** metadata update request message](#partial-**endpoinnt**-metadata-update-request)